src/chainstate/coordinator/tests.rs

```diff
@@ -3236,6 +3236,181 @@ fn test_stx_transfer_btc_ops() {
+            make_genesis_block_with_recipients(
+                &sort_db,
+                &mut chainstate,
+                &parent,
+                miner,
+                10000,
+                vrf_key,
+                ix as u32,
+                None,
+            )
+        } else {
+            make_stacks_block_with_recipients(
+                &sort_db,
+                &mut chainstate,
+                &b,
+                &parent,
+                burnchain_tip.block_height,
+                miner,
+                1000,
+                vrf_key,
+                ix as u32,
+                None,
+            )
+        };
+
+        let expected_winner = good_op.txid();
+        let mut ops = vec![good_op];
+        let peg_wallet_address = PoxAddress::Addr32(false, PoxAddressType32::P2TR, [0; 32]);
+
+        if ix == 0 {
+            // add a peg-in op
+            ops.push(BlockstackOperationType::PegIn(PegInOp {
+                recipient: stacker.into(),
+                peg_wallet_address,
+                amount: 1337,
+                txid: next_txid(),
+                vtxindex: 5,
+                block_height: 0,
+                burn_header_hash: BurnchainHeaderHash([0; 32]),
+            }));
+        } else if ix == 1 {
+            // shouldn't be accepted -- amount must be positive
+            ops.push(BlockstackOperationType::PegIn(PegInOp {
+                recipient: stacker.into(),
+                peg_wallet_address,
+                amount: 0,
+                txid: next_txid(),
+                vtxindex: 5,
+                block_height: 0,
+                burn_header_hash: BurnchainHeaderHash([0; 32]),
+            }));
+        }
+
+        let burnchain_tip = burnchain.get_canonical_chain_tip().unwrap();
+        produce_burn_block(
+            &b,
+            &mut burnchain,
+            &burnchain_tip.block_hash,
+            ops,
+            vec![].iter_mut(),
+        );
+
+        burnchain_block_hashes.push(burnchain_tip.block_hash);
+        // handle the sortition
+        coord.handle_new_burnchain_block().unwrap();
+
+        let tip = SortitionDB::get_canonical_burn_chain_tip(sort_db.conn()).unwrap();
+        assert_eq!(&tip.winning_block_txid, &expected_winner);
+
+        // load the block into staging
+        let block_hash = block.header.block_hash();
+
+        assert_eq!(&tip.winning_stacks_block_hash, &block_hash);
+        stacks_blocks.push((tip.sortition_id.clone(), block.clone()));
+
+        preprocess_block(&mut chainstate, &sort_db, &tip, block);
+
+        // handle the stacks block
+        coord.handle_new_stacks_block().unwrap();
+    }
+
+    let total_number_of_peg_in_ops = burnchain_block_hashes
+        .into_iter()
+        .flat_map(|block_hash| {
+            SortitionDB::get_peg_in_ops(&sort_db.conn(), &block_hash)
+                .expect("Failed to get peg in ops")
+        })
+        .count();
+
+    assert_eq!(total_number_of_peg_in_ops, 1);
```

Comment:
Please also check that it's the one you expect.

Reply:
Done :heavy_check_mark: